
```{r}
BiocManager::install("org.Hs.eg.db")
```


Import libraries
```{r}
library(DESeq2)
library(tidyverse)
library(ggalt)
library(ggrepel)
library(pheatmap)
library(RColorBrewer)
library(BiocGenerics)
library(org.Hs.eg.db)
library(AnnotationDbi)
library(EnhancedVolcano)
library(clusterProfiler)
library(enrichplot)
library(DOSE)
```

Import data
```{r}
Counts <- read.delim("data/count_matrix.txt", header = TRUE, row.names = 1, sep = "\t", dec = ".")
```

check data
```{r}
ggplot(Counts) +
  geom_histogram(aes(x = healthy1), stat = "bin", bins = 200) + 
  xlim(-5, 1000)  +
  xlab("Raw expression counts") +
  ylab("Number of genes")
```
Remove rows with low counts
```{r}
Counts <- Counts[rowSums(Counts) > 50, ]
```

Create condition for samples
```{r}
condition <- factor(c(rep("Healthy", 3), rep("Patient", 6)))
```

Create a metadata
```{r}
coldata <- data.frame(row.names = colnames(Counts), condition)
```

Check the metadata and count data columns name
```{r}
colnames(Counts) %in% rownames(coldata)
```
Create a DESeqDataSet object (dds) that combines the count data and the sample metadata
```{r}
dds <- DESeqDataSetFromMatrix(countData = Counts,
                              colData = coldata,
                              design = ~ condition)
dds <- DESeq(dds)
```

Create dispersion plot
The plotDispEsts function is a valuable tool for understanding the dispersion of your RNA-seq data and evaluating whether the DESeq2 model assumptions are appropriate. By visualizing the relationship between mean and dispersion, you can gain insights into the data quality and make informed decisions about the subsequent differential expression analysis.
```{r}
plotDispEsts(dds, main="Dispersion plot")
```

Perform variance-stabilizing transformation (VST) on normalized RNA-seq count data 
```{r}
vsdata <- vst(dds, blind = FALSE)
```

Create a PCA plot
```{r}
plotPCA(vsdata, intgroup = "condition")
```

Modified plot
```{r}

vsdata <- vst(dds, blind = FALSE)
pcaData <- plotPCA(vsdata, intgroup = "condition", returnData = TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))
```

PCA plot
```{r}
png("figures/PCA_plot.png", res = 250, height = 1500, width = 2000 )
ggplot(pcaData, aes_string(x = "PC1", y = "PC2", color = "group")) +
  geom_point(size = 5) +
  stat_ellipse(type = "norm", level = 0.85, alpha = 0.5) +
  labs(x = paste0("PC1: ", percentVar[1], "% variance"),
       y = paste0("PC2: ", percentVar[2], "% variance"),
       color = "Sample Group") +
  theme_classic() +
  theme(legend.position = "right")
dev.off()
```

Log transformation

```{r}
rld <- rlog(dds, blind=TRUE)
#plotPCA(rld) # PCA plot with the log transformed value
rld_mat <- assay(rld)    
rld_cor <- cor(rld_mat)   
```

Plot heatmap
```{r}
pheatmap(rld_cor,
         cluster_rows = F,
          cluster_cols = F)
```
The results function in DESeq2 is used to extract the results of the differential expression analysis from a DESeqDataSet object.
It provides a table of statistics for each gene, including the log2 fold changes, p-values, and adjusted p-values.
```{r}
contrast_gr <- c("condition", "Healthy", "Patient")
res <- results(dds, contrast=contrast_gr)
```

Plotting
```{r}
plotMA(res)
```
```{r}
sigsOE <- res %>%
  data.frame() %>%
  rownames_to_column(var="gene") %>% 
  as_tibble()
#%>%
  #filter(padj < 0.05 & abs(log2FoldChange) > 1)
```


```{r}
normalized_counts <- counts(dds, normalized=TRUE)
```


```{r}
normalized_counts <- normalized_counts %>% 
  data.frame() %>%
  rownames_to_column(var="gene") %>% 
  as_tibble()
```


```{r}
norm_OE <- normalized_counts %>% 
  filter(gene %in% sigsOE$gene) %>% 
  data.frame() %>%
  column_to_rownames(var = "gene") 
```

```{r}
pheatmap(norm_OE,
         cluster_rows = FALSE,
         cluster_cols = F,
         show_rownames = FALSE)
```

Convert to dataframe
```{r}
sigsOE_data <- as.data.frame(sigsOE)
```


```{r}
sigsOE_genes <- sigsOE_data %>%
  mutate(gene_symbol = mapIds(org.Hs.eg.db, 
                              keys = gene, 
                              column = "SYMBOL", 
                              keytype = "ENSEMBL", 
                              multiVals = "first"))
sigsOE_genes <- na.omit(sigsOE_genes)
```

Create volcano plot
```{r}
png("figures/Volcano Plot.png", res = 250, width = 3300, height = 3300)
EnhancedVolcano(sigsOE_genes,
                x = "log2FoldChange",
                y = "padj",
                title = 'Healthy versus COVID-19',
                lab = sigsOE_genes$gene_symbol,
                pCutoff = 10e-17,
                FCcutoff = 1.5,
                pointSize = 5.0,
                labSize = 6.0,
                colAlpha = 1,
                legendLabels=c('Not sig.','Log (base 2) FC','p-value','p-value & Log (base 2) FC'),
                legendPosition = 'right',
                legendLabSize = 16,
                legendIconSize = 5.0,
                drawConnectors = TRUE,
                widthConnectors = 0.75)
dev.off()
```


# Gene ontology (GO)
```{r}
gene_annotation <- enrichGO(gene = sigsOE_genes$gene, OrgDb = "org.Hs.eg.db", keyType = "ENSEMBL", ont = "BP")

```

```{r}
View(as.data.frame(gene_annotation))
```

Bar Plot
```{r}
png("figures/Barplot.png", res = 250, width = 2200, height = 3000)
plot(barplot(gene_annotation, showCategory = 40))
dev.off()
```
Dot plot
```{r}
png("figures/Dotplot_ORA.png", res = 250, width = 2200, height = 3300)
dotplot(gene_annotation, showCategory=40) + ggtitle("Dotplot for ORA")
dev.off()
```

